src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java

```diff
@@ -78,25 +77,33 @@ protected Optional<Issue> createIssue(final Matcher matcher, final LookaheadStre
                     .setSeverity(Severity.WARNING_NORMAL)
                     .buildOptional();
         }
-        if (StringUtils.isNotEmpty(matcher.group(10))) {
-            return builder.setLineStart(matcher.group(5))
+        if (StringUtils.isNotEmpty(matcher.group(5))) {
+            builder.setLineStart(matcher.group(5))
                     .setColumnStart(matcher.group(6))
-                    .setCategory(matcher.group(9))
-                    .setType(matcher.group(10))
-                    .setMessage(matcher.group(11))
-                    .setSeverity(Severity.guessFromString(matcher.group(8)))
+                    .setLineEnd(matcher.group(7))
+                    .setColumnEnd(matcher.group(8));
+        }
+        if (StringUtils.isNotEmpty(matcher.group(9))) {
+            builder.setLineStart(matcher.group(9))
+                    .setColumnStart(matcher.group(11))
+                    .setLineEnd(matcher.group(10))
+                    .setColumnEnd(matcher.group(12));
+        }
```

Comment:
I thought the idea was to have one parser for each of these blocks? The new implementation is better, but I think it can be improved by having a separate parser for each disjunct pattern.